debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,110 +150,213 @@ public String getName() {
+                }
+            case "lseg":
+                try {
+                    return new PGlseg(rawValue.asString());
                 } catch (final SQLException e) {
                     LOGGER.error("Failed to parse point {}, {}", rawValue.asString(), e);
                     throw new ConnectException(e);
                 }
             case "money":
                 try {
-                    return rawValue.isNotNull() ? new PGmoney(rawValue.asString()).val : null;
+                    return new PGmoney(rawValue.asString()).val;
                 } catch (final SQLException e) {
                     LOGGER.error("Failed to parse money {}, {}", rawValue.asString(), e);
                     throw new ConnectException(e);
                 }
-            case "interval":
+            case "path":
                 try {
-                    return rawValue.isNotNull() ? new PGInterval(rawValue.asString()) : null;
+                    return new PGpath(rawValue.asString());
                 } catch (final SQLException e) {
                     LOGGER.error("Failed to parse point {}, {}", rawValue.asString(), e);
                     throw new ConnectException(e);
                 }
-            case "_int2":
-            case "_int4":
-            case "_int8":
-            case "_text":
-            case "_numeric":
-            case "_float4":
-            case "_float8":
-            case "_bool":
-            case "_date":
-            case "_time":
-            case "_timetz":
-            case "_timestamp":
-            case "_timestamptz":
-            case "_bytea":
-            case "_varchar":
-            case "_oid":
-            case "_bpchar":
-            case "_money":
-            case "_name":
-            case "_interval":
-            case "_char":
-            case "_varbit":
-            case "_uuid":
-            case "_xml":
-            case "_point":
-            case "_jsonb":
-            case "_json":
-            case "_ref_cursor":
+            case "point":
                 try {
-                    if (rawValue.isNull()) {
-                        return null;
-                    }
-                    final String dataString = rawValue.asString();
-                    PgArray arrayData = new PgArray(connection.get(), Oid.valueOf(columnType.substring(1) + "_array"), dataString);
-                    Object deserializedArray = arrayData.getArray();
-                    return Arrays.asList((Object[])deserializedArray);
+                    return new PGpoint(rawValue.asString());
+                } catch (final SQLException e) {
+                    LOGGER.error("Failed to parse point {}, {}", rawValue.asString(), e);
+                    throw new ConnectException(e);
                 }
-                catch (SQLException e) {
-                    LOGGER.warn("Unexpected exception trying to process PgArray column '{}'", columnName, e);
+            case "polygon":
```

Comment:
Do we have a test somewhere for these geometric types? I don't think they where supported via DecoderBufs (with the exception of `POINT`).

Reply:
They're builtin types, but I think there's only a test for point. Not sure if they're used much/at-all, IME (as a been-doing-this-way-too-long geo developer) everyone uses PostGIS :)